### PR TITLE
feat: [CDS-50514]: Applying on RBAC on services overview.

### PR DESCRIPTION
**Pull request by boopesh007**
## Describe your changes
Applied RBAC on the Instances query which will filter only the instances to which the user has access. (Whichever the services the user has READ access)

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)

### CODE REVIEW

**Review by Fernando-Dourado — CHANGES_REQUESTED** [review]
> Reviewed 3 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)* status: 4 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/UsageRestrictionsServiceImpl.java` line 353 at r1](https://reviewable.io/reviews/harness/harness-core/45345#-NR076kv0_wDZcGJ621q:-NR076kv0_wDZcGJ621r:bmxl1e7) ([raw file](https://github.com/harness/harness-core/blob/f9a56d57b22ecd397bea50fd12410185ae3e0810/400-rest/src/main/java/software/wings/service/impl/UsageRestrictionsServiceImpl.java#L353)):*
> ```Java
> 
>   @Override
>   public Map<String, Set<String>> getAppSvcMapFromUserPermissions(
> ```

Please, add unit tests to cover the new scenarios.

File: 400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java [hunk 1]
```diff
@@ -639,6 +642,15 @@ private Query<Instance> getInstanceQueryAtTime(String accountId, List<String> ap
     } else {
       query = getInstanceQuery(accountId, appIds, false, timestamp);
     }
+    if (featureFlagService.isEnabled(SPG_SERVICES_OVERVIEW_RBAC, accountId)) {
+      UserRequestContext userRequestContext = UserThreadLocal.get().getUserRequestContext();
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java:646]: UserThreadLocal.get()!=null

Comment (abhinav-harness) [reply 400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java:646]: private boolean hasUserContext() {
    User user = UserThreadLocal.get();
    return user != null && user.getUserRequestContext() != null;
  }

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/UsageRestrictionsServiceImpl.java` line 353 at r1](https://reviewable.io/reviews/harness/harness-core/45345#-NR076kv0_wDZcGJ621q:-NR2mq-cBkdEzgIRxTlM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f9a56d57b22ecd397bea50fd12410185ae3e0810/400-rest/src/main/java/software/wings/service/impl/UsageRestrictionsServiceImpl.java#L353)):*
<details><summary><i>Previously, Fernando-Dourado (Fernando Dourado) wrote…</i></summary><blockquote>

Please, add unit tests to cover the new scenarios.
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java` line 646 at r1](https://reviewable.io/reviews/harness/harness-core/45345#-NQxuEP86icmP9DnICLR-r1-646:-NR2mqwj0AOig1xz5YsJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f9a56d57b22ecd397bea50fd12410185ae3e0810/400-rest/src/main/java/software/wings/service/impl/instance/DashboardStatisticsServiceImpl.java#L646)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

  private boolean hasUserContext() {
    User user = UserThreadLocal.get();
    return user != null && user.getUserRequestContext() != null;
  }
</blockquote></details>

Done.

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)* status: 7 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 2 of 4 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by VikasMaddukuriHarness — APPROVED** [review]
> Reviewed 2 of 4 files at r1, 3 of 4 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45345)* status: :shipit: complete! ✓

**Pull request merged**